### ISSUE TITLE
Error while training from scratch

### ISSUE THREAD

**Issue by egor-bogomolov** [tech-srl/code2vec#2]
Hello! I've tried to train a model from scratch and received following error:

`Average loss at batch 12300: 0.001828, 	throughput: 1302 samples/sec`
`Number of waiting examples in queue: 300000`
`2018-10-24 00:21:02.697593: E tensorflow/stream_executor/cuda/cuda_driver.cc:868] failed to alloc 4294967296 bytes on host: CUDA_ERROR_INVALID_VALUE: invalid argument`
`2018-10-24 00:21:02.787920: W ./tensorflow/core/common_runtime/gpu/cuda_host_allocator.h:44] could not allocate pinned host memory of size: 4294967296`

It seems like something happened right at the end of first epoch.
I'm training a model in cloud with Tesla K80 and ~12 GB of RAM.
Do you have any idea on what could've caused an error?

**Comment by urialon** [comment]
Hi Egor,

1. Did you use our preprocessed data?
2. Can you please share your:
 a) Cuda version (nvcc --version)
 b) cudnn version (cat /usr/local/cuda/include/cudnn.h | grep CUDNN_MAJOR -A 2)
 c) TensorFlow version (python3 -c 'import tensorflow as tf; print(tf.__version__)')
3. Can you try to decrease the batch size by half (under common.py line 13, set the value of config.BATCH_SIZE to be 512)?

**Comment by egor-bogomolov** [comment]
1. Yes, I downloaded and extracted your data.
2. My versions are:
    a) release 9.2, V9.2.148
    b) There is no such file :(
    c) 1.12.0-rc1
3. Sure, will try it. But it will take another ~3 hours :)

**Comment by urialon** [comment]
Try to install cudnn [https://docs.nvidia.com/deeplearning/sdk/cudnn-install/index.html](https://docs.nvidia.com/deeplearning/sdk/cudnn-install/index.html), it is good to have for any TensorFlow run. 
I'm trying to validate as well using your versions.

**Comment by urialon** [comment]
Sorry, try to check CUDNN version using: `cat /usr/include/x86_64-linux-gnu/cudnn_v*.h | grep CUDNN_MAJOR -A 2`

you might have it already.

**Comment by egor-bogomolov** [comment]
Still nothing found.

**Comment by urialon** [comment]
Then try to install cuDNN. In the meantime I'm running on the same GPU and versions as you.

**Comment by egor-bogomolov** [comment]
Hello, I will install cuDNN and try to run training a bit later in the evening and will write you the results.

Thanks a lot for your help :+1:

**Comment by egor-bogomolov** [comment]
`train.sh: line 17:   462 Killed                  python3 -u code2vec.py --data ${data} --test ${test_data} --save ${model_dir}/saved_model`

It looks like I don't have enough memory. How much memory does computation take in your case? How much RAM did you have?

**Comment by urialon** [comment]
Egor, how is it going?
Did you try any of these potential solutions? Does the problem still occur?
Otherwise, can I close the issue?

**Comment by egor-bogomolov** [comment]
Oops, forgot to write the result here. I've made the dictionary smaller and everything worked fine. Afterwards I tried to reproduce the error again and wasn't able to do it.
After all I've reproduced your results from the paper  without any problems. Looks like installing cuDNN was a correct first step :)
Thanks a lot for your help, you can close the issue for sure.
